pkg/compiler/vars.go

```diff
@@ -0,0 +1,63 @@
+package compiler
+
+type varScope struct {
+	localsCnt int
+	argCnt    int
+	arguments map[string]int
+	locals    []map[string]int
+}
+
+func newVarScope() varScope {
+	return varScope{
+		arguments: make(map[string]int),
+	}
+}
+
+func (c *varScope) newScope() {
+	c.locals = append(c.locals, map[string]int{})
+}
+
+func (c *varScope) dropScope() {
+	c.locals = c.locals[:len(c.locals)-1]
+}
+
+func (c *varScope) getVarIndex(name string) (varType, int) {
+	if i, ok := c.arguments[name]; ok {
```

Comment:
What about this?
```go
func some(a int) int {
    if a > 42 {
        a := 24
        return a
    }
    return a
}
```

Reply:
This will work, though if we don't return inside `if`, result will be incorrect. Moving this `if` down seems enough. Added a test.